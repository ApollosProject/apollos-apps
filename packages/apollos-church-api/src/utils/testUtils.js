import getContext from 'apollos-church-api/src/getContext';
import getDataSources from 'apollos-church-api/src/getDataSources';
import { KeyValueCache } from 'apollo-server-caching';

export function getTestContext(req) {
  const context = getContext(req);

  const dataSources = getDataSources();
  // Apollo Server does this internally.
  Object.values(dataSources).forEach((dataSource) => {
    if (dataSource.initialize) {
      dataSource.initialize({ context, cache: KeyValueCache });
    }
  });
  context.dataSources = dataSources;
  return context;
}

export const buildGetMock = (response, dataSource) => {
  const get = jest.fn();
  if (Array.isArray(response) && Array.isArray(response[0])) {
    response.forEach((responseVal) => {
      get.mockReturnValueOnce(
        new Promise((resolve) => resolve(dataSource.normalize(responseVal)))
      );
    });
  }
  get.mockReturnValue(
    new Promise((resolve) => resolve(dataSource.normalize(response)))
  );
  return get;
};
