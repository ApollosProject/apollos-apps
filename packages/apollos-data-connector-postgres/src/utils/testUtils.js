/* eslint-disable import/prefer-default-export */
import createMigrationRunner from '../postgres/performMigrations';
import { sequelize } from '../postgres';

export const setupPostgresTestEnv = async (models = [], context) => {
  if (context.church.slug !== 'global') {
    await setupPostgresTestEnv(models, { church: { slug: 'global' } });
  }

  try {
    // If we don't have a set of migrations.
    // Hacky way of saying - have the migrations been run?
    await sequelize
      .getQueryInterface()
      .sequelize.query('select * from "SequelizeMeta"');
  } catch (e) {
    const migrationRunner = await createMigrationRunner({
      logger: null,
    });
    await migrationRunner.up();
  }

  await Promise.all(
    models.map((m) => m.models.createModel && m.models.createModel(context))
  );

  await Promise.all(
    models.map((m) => m.models.setupModel && m.models.setupModel(context))
  );
};
