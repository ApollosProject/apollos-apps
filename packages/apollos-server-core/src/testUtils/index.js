/* eslint-disable import/prefer-default-export */
import { makeExecutableSchema } from 'apollo-server';
import { testSchema } from '@apollosproject/data-schema';
import { KeyValueCache } from 'apollo-server-caching';
import { createApolloServerConfig } from '..';

export const createTestHelpers = (models) => {
  const serverConfig = createApolloServerConfig(models);
  const getContext = (req) => {
    const testContext = serverConfig.context(req);
    const testDataSources = serverConfig.dataSources();
    // Apollo Server does this internally.
    Object.values(testDataSources).forEach((dataSource) => {
      if (dataSource.initialize) {
        dataSource.initialize({ context: testContext, cache: KeyValueCache });
      }
    });
    testContext.dataSources = testDataSources;
    return testContext;
  };
  const getSchema = (schemas = []) =>
    makeExecutableSchema({
      typeDefs: [...serverConfig.schema, ...schemas, testSchema],
      resolvers: serverConfig.resolvers,
      resolverValidationOptions: {
        requireResolversForResolveType: false,
      },
    });

  return { getSchema, getContext };
};
