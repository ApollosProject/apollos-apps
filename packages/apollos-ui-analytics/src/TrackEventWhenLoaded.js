import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { AnalyticsConsumer } from './Provider';

class TrackEventWhenLoaded extends PureComponent {
  componentDidMount() {
    if (this.props.loaded || this.props.isLoading === false) {
      this.track();
    }
  }

  componentDidUpdate(prevProps) {
    if (
      (!prevProps.loaded && this.props.loaded) ||
      (prevProps.isLoading && !this.props.isLoading)
    ) {
      this.track();
    }
  }

  get trackClient() {
    return this.props.track;
  }

  track() {
    const { eventName, properties } = this.props;
    return this.trackClient({ eventName, properties });
  }

  render() {
    return null;
  }
}

TrackEventWhenLoaded.propTypes = {
  loaded: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  eventName: PropTypes.string.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  properties: PropTypes.any,
  // eslint-disable-next-line react/forbid-prop-types
  track: PropTypes.func,
};

const TrackEventWhenLoadedConnected = (props) => (
  <AnalyticsConsumer>
    {({ track }) => <TrackEventWhenLoaded {...props} track={track} />}
  </AnalyticsConsumer>
);

export default TrackEventWhenLoadedConnected;
