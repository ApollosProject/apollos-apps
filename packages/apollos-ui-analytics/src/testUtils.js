import React from 'react';

import { Providers } from '@apollosproject/ui-kit';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';
import AnalyticsProvider from './Provider';

const cache = new InMemoryCache();

export const trackMock = jest.fn();

const client = new ApolloClient({
  link: createHttpLink({ fetch: jest.fn() }),
  cache,
  resolvers: {},
});

export const wait = (time) =>
  new Promise((resolve) => setTimeout(resolve, time));

// eslint-disable-next-line
export const ProviderStack = ({ children }) => (
  <ApolloProvider client={client}>
    <Providers>
      <AnalyticsProvider trackFunctions={[trackMock]}>
        {children}
      </AnalyticsProvider>
    </Providers>
  </ApolloProvider>
);
