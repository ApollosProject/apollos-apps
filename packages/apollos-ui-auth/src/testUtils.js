/* eslint-disable import/prefer-default-export */
import React from 'react';
import { Providers as UIProviders } from '@apollosproject/ui-kit';
import { MockedProvider } from 'react-apollo/test-utils';

// eslint-disable-next-line
export const Providers = ({ children, ...props }) => (
  <UIProviders {...props}>
    <MockedProvider {...props}>{children}</MockedProvider>
  </UIProviders>
);
