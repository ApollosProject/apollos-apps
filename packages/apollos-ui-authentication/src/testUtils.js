import React from 'react';
import PropTypes from 'prop-types';
import { Providers as UIProviders } from '@apollosproject/ui-kit';
import { MockedProvider } from '@apollo/client/testing';

export const Providers = ({ children, ...props }) => (
  <UIProviders {...props}>
    <MockedProvider {...props}>{children}</MockedProvider>
  </UIProviders>
);

Providers.propTypes = {
  children: PropTypes.node,
};
