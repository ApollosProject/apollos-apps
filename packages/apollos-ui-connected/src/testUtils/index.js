/* eslint-disable import/prefer-default-export */
import { createStackNavigator, createAppContainer } from 'react-navigation';
import { IntrospectionFragmentMatcher } from 'apollo-cache-inmemory';
import apolloStorybookDecorator from 'apollo-storybook-react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';
import { Providers as UIProviders } from '@apollosproject/ui-kit';
import { MockedProvider } from 'react-apollo/test-utils';
import possibleTypesJson from './fragmentTypes.json';
import typeDefs from './typeDefsMock';

const renderWithApolloData = async (component, existingTree) => {
  const tree = existingTree || renderer.create(component);
  await wait(1);
  tree.update(component);
  return tree;
};

// eslint-disable-next-line
const Providers = ({ children, ...props }) => (
  <UIProviders {...props}>
    <MockedProvider {...props}>{children}</MockedProvider>
  </UIProviders>
);

const ApolloStorybookDecorator = ({
  mocks,
  additionalSchema = [],
  possibleTypes = {},
}) => {
  // We can use this way simpler code long term when we upgrade to Apollo 3.
  // const finalPossibleTypes = possibleTypes;
  // possibleTypesJson.__schema.types.forEach((supertype) => {
  //   if (supertype.possibleTypes) {
  //     finalPossibleTypes[supertype.name] = [
  //       ...supertype.possibleTypes.map((subtype) => subtype.name),
  //       ...(possibleTypes[supertype.name] || []),
  //     ];
  //   }
  // });
  const finalPossibleTypes = possibleTypesJson;
  Object.keys(possibleTypes).forEach((key) => {
    const jsonTypeIndex = possibleTypesJson.__schema.types.findIndex(
      ({ name }) => name === key
    );
    const newTypePossibleTypes = {
      ...possibleTypesJson.__schema.types[jsonTypeIndex],
      possibleTypes: [
        ...possibleTypesJson.__schema.types[jsonTypeIndex].possibleTypes,
        ...possibleTypes[key].map((type) => ({ name: type })),
      ],
    };
    possibleTypesJson.__schema.types[jsonTypeIndex] = newTypePossibleTypes;
  });
  const fragmentMatcher = new IntrospectionFragmentMatcher({
    introspectionQueryResultData: finalPossibleTypes,
  });

  return apolloStorybookDecorator({
    typeDefs: [typeDefs, ...additionalSchema],
    mocks,
    resolverValidationOptions: {
      requireResolversForResolveType: false,
    },
    cacheOptions: { fragmentMatcher },
  });
};

const WithReactNavigator = (Component) => {
  const AppNavigator = createStackNavigator({
    Home: {
      screen: () => Component,
      navigationOptions: {
        headerMode: 'none',
        header: null,
      },
    },
  });

  const AppContainer = createAppContainer(AppNavigator);
  return <AppContainer />;
};
export {
  renderWithApolloData,
  Providers,
  ApolloStorybookDecorator,
  WithReactNavigator,
};
