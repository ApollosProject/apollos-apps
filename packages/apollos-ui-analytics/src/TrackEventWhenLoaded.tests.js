import React from 'react';
import renderer from 'react-test-renderer';
import TrackEventWhenLoaded from './TrackEventWhenLoaded';
import { ProviderStack, wait, trackMock } from './testUtils';

describe('The TrackEventWhenLoaded component', () => {
  afterEach(() => {
    trackMock.mockClear();
  });
  it('should track an event if mounted with loaded=true', async () => {
    renderer.create(
      <ProviderStack>
        <TrackEventWhenLoaded
          loaded
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
        />
      </ProviderStack>
    );
    await wait(1000);
    expect(trackMock.mock.calls).toMatchSnapshot();
  });

  it('should track an event if mounted with isLoading=false', async () => {
    renderer.create(
      <ProviderStack>
        <TrackEventWhenLoaded
          isLoading={false}
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
        />
      </ProviderStack>
    );
    await wait(1000);
    expect(trackMock.mock.calls).toMatchSnapshot();
  });

  it('should track an event if mounted with loaded=false and then switching to true', async () => {
    const tree = renderer.create(
      <ProviderStack>
        <TrackEventWhenLoaded
          loaded={false}
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
          track={trackMock}
          key="tracker"
        />
      </ProviderStack>
    );
    await wait(1000);
    expect(trackMock.mock.calls).toMatchSnapshot('mounted');
    tree.update(
      <ProviderStack>
        <TrackEventWhenLoaded
          loaded={false}
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
          track={trackMock}
          key="tracker"
        />
      </ProviderStack>
    );
    await wait(1000);
    expect(trackMock.mock.calls).toMatchSnapshot('first update, not loaded');
    tree.update(
      <ProviderStack>
        <TrackEventWhenLoaded
          loaded
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
          track={trackMock}
          key="tracker"
        />
      </ProviderStack>
    );
    await wait(1000);
    expect(trackMock.mock.calls).toMatchSnapshot('second update, loaded');
  });

  it('should track an event if mounted with isLoading=true and then switching to false', async () => {
    const tree = renderer.create(
      <ProviderStack>
        <TrackEventWhenLoaded
          isLoading
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
          track={trackMock}
          key="tracker"
        />
      </ProviderStack>
    );
    await wait(1000);
    expect(trackMock.mock.calls).toMatchSnapshot('mounted');
    tree.update(
      <ProviderStack>
        <TrackEventWhenLoaded
          isLoading
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
          track={trackMock}
          key="tracker"
        />
      </ProviderStack>
    );
    await wait(1000);
    expect(trackMock.mock.calls).toMatchSnapshot('first update, not loaded');
    tree.update(
      <ProviderStack>
        <TrackEventWhenLoaded
          isLoading={false}
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
          track={trackMock}
          key="tracker"
        />
      </ProviderStack>
    );
    await wait(1000);
    expect(trackMock.mock.calls).toMatchSnapshot('second update, loaded');
  });
});
