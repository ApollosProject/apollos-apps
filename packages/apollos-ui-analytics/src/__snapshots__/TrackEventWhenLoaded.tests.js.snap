// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TrackEventWhenLoaded component should track an event if mounted with isLoading=false 1`] = `
Array [
  Array [
    Object {
      "eventName": "Track Something",
      "properties": Object {
        "ThingId": "1234",
      },
    },
  ],
]
`;

exports[`The TrackEventWhenLoaded component should track an event if mounted with isLoading=true and then switching to false: first update, not loaded 1`] = `Array []`;

exports[`The TrackEventWhenLoaded component should track an event if mounted with isLoading=true and then switching to false: mounted 1`] = `Array []`;

exports[`The TrackEventWhenLoaded component should track an event if mounted with isLoading=true and then switching to false: second update, loaded 1`] = `
Array [
  Array [
    Object {
      "eventName": "Track Something",
      "properties": Object {
        "ThingId": "1234",
      },
    },
  ],
]
`;

exports[`The TrackEventWhenLoaded component should track an event if mounted with loaded=false and then switching to true: first update, not loaded 1`] = `Array []`;

exports[`The TrackEventWhenLoaded component should track an event if mounted with loaded=false and then switching to true: mounted 1`] = `Array []`;

exports[`The TrackEventWhenLoaded component should track an event if mounted with loaded=false and then switching to true: second update, loaded 1`] = `
Array [
  Array [
    Object {
      "eventName": "Track Something",
      "properties": Object {
        "ThingId": "1234",
      },
    },
  ],
]
`;

exports[`The TrackEventWhenLoaded component should track an event if mounted with loaded=true 1`] = `
Array [
  Array [
    Object {
      "eventName": "Track Something",
      "properties": Object {
        "ThingId": "1234",
      },
    },
  ],
]
`;
